src/utils/metric/metric.ts

```diff
@@ -0,0 +1,86 @@
+/*
+	Copyright 2020 Dynatrace LLC
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+export interface SummaryValue {
+	min: number;
+	max: number;
+	sum: number;
+	count: number;
+}
+
+export interface Dimension {
+	key: string;
+	value: string;
+}
+
+export interface Metric {
+	serialize(): string;
+}
+
+export abstract class BaseMetric<T> implements Metric {
+	protected key: string;
+	protected dimensions: Dimension[];
+	protected value: T;
+	protected timestamp?: number;
+
+	constructor(key: string, dimensions: Dimension[], value: T, timestamp?: number) {
+		this.key = key;
+		this.dimensions = dimensions;
+		this.value = value;
+		this.timestamp = timestamp;
+	}
+
+	public serialize(): string {
+		let line = this.key;
```

Comment:
Is it possible for the key to be empty here if someone were to pass an empty key or a key that consists only of invalid characters?

Reply:
I saw the tests down below and it seems as if the metric creation would already prohibit invalid characters, so this is probably solved.